k-distribution/src/main/scripts/lib/opam/compilers/4.03.0/4.03.0+k/files/opam-compiler.patch

```diff
@@ -20,6 +20,37 @@ index 7d54160..5a51708 100644
      let fail,jmps =
        scan_def
          []
+diff --git a/byterun/debugger.c b/byterun/debugger.c
+index 5e61584..55ccf62 100644
+--- a/byterun/debugger.c
++++ b/byterun/debugger.c
+@@ -161,6 +161,7 @@ static void winsock_cleanup(void)
+ void caml_debugger_init(void)
+ {
+   char * address;
++  size_t a_len;
+   char * port, * p;
+   struct hostent * host;
+   int n;
+@@ -188,11 +189,16 @@ void caml_debugger_init(void)
+     /* Unix domain */
+     sock_domain = PF_UNIX;
+     sock_addr.s_unix.sun_family = AF_UNIX;
++    a_len = strlen(address);
++    if (a_len >= sizeof(sock_addr.s_unix.sun_path)) {
++      caml_fatal_error("Debug socket path length exceeds maximum permitted length");
++    }
+     strncpy(sock_addr.s_unix.sun_path, address,
+-            sizeof(sock_addr.s_unix.sun_path));
++            sizeof(sock_addr.s_unix.sun_path) - 1);
++    sock_addr.s_unix.sun_path[sizeof(sock_addr.s_unix.sun_path) - 1] = '\0';
```

Comment:
If you fail with `fatal_error` above, why is this change still needed? Is  this just for safety, in case the above is just a warning?

Reply:
It's not really my code, I'm just backporting a change from the trunk of ocaml.